tests/test_main.py

```diff
@@ -103,3 +104,17 @@ def mocked_main(*args):
 
     with pytest.raises(FakeFailureError):
         xonsh.main.main()
+
+
+def test_xonsh_failback_script_from_file(shell, monkeypatch):
```

Comment:
Not an issue of this PR, but if this new test is run based on the old version of `xonsh/main.py` file.

This is pass (20 passed)
```
py.test tests/test_main.py
```

And this will fail (1 failed, 19 passed)
```
py.test -s tests/test_main.py
```

quite confused to me.